getting-started/nextjs.mdx

```diff
@@ -0,0 +1,275 @@
+export const config = {
+  matcher: [
+      // REQUIRED: Match all request paths that start with /api/auth/
+      '/api/auth/(.*)',
+      // OPTIONAL: Don't match any static assets
+      '/((?!_next/static|_next/image|favicon.ico).*)',
+  ],
+}
+```
+
+This will wrap and protect any routes that start with `/api/auth/`.
+
+## Basic Usage
+
+Now that our project has been set up, below are a few examples of basic auth functionality 
+using PropelAuth. As a reminder, Server components cannot have any user interactivity, so be 
+sure to keep your desired functionality in mind when choosing which components to use.
+
+### Get the user in Server Components
+
+<Tabs>
+  <Tab title="AppRouter Version">
+		```tsx
+		import {getUser} from "@propelauth/nextjs/server/app-router";
+
+		const WelcomeMessage = async () => {
+			const user = await getUser()
+			
+			if (user) {
+				return <div>Hello {user.firstName}!</div>
+			} else {
+				return <div>Please log in to be welcomed</div>
+			}
+		}
+		```
+		```tsx
+		import {getUser} from "@propelauth/nextjs/server/app-router";
+
+		const WelcomeMessage = async () => {
+			// If the user is not logged in, they will be redirected to the login page
+			const user = await getUserOrRedirect()
+			
+			return <div>Hello {user.firstName}!</div>
+		}
+		```
+  </Tab>
+	<Tab title="Pages Version">
+
+		```tsx
+		import {GetServerSideProps, InferGetServerSidePropsType} from "next";
+		import {getUserFromServerSideProps} from "@propelauth/nextjs/server/pages";
+		import {User} from "@propelauth/nextjs/client";
+
+		export default function WelcomeMessage({userJson}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+			// Deserialize the user from the JSON string so you can call functions like user.getOrg()
+			const user = User.fromJSON(userJson)
+			return <div>Hello, {user.firstName}</div>
+		}
+
+		export const getServerSideProps: GetServerSideProps = async (context) => {
+			const user = await getUserFromServerSideProps(context)
+			if (!user) {
+				return {redirect: {destination: '/api/auth/login', permanent: false}}
+			}
+			return { props: {userJson: JSON.stringify(user) } }
+		}
+		```
+  </Tab>
+</Tabs>
+
+
+### Get the user in Client Components
+
+```tsx
+"use client";
+
+import {useUser} from "@propelauth/nextjs/client";
+
+const WelcomeMessage = () => {
+  const {loading, user} = useUser()
+  if (loading) {
+    return <div>Loading...</div>
+  } else if (user) {
+    return <div>Hello {user.email}!</div>
+  } else {
+    return <div>Please log in to be welcomed</div>
+  }
+}
+```
+
+### Checking organization membership / RBAC
+
+Note that this works on both the client and server's `User` object, but the below example is on the server.
+
+If you are curious where the organization information comes from, check out our documentation on 
+[organizations](https://docs.propelauth.com/overview/organizations?utm_source=github&utm_medium=library&utm_campaign=nextjs). 
+The quick answer is:
```

Comment:
I don't think we need this here since the other docs page has them. The github repo was being extra verbose